### ISSUE TITLE
Test Github actions calling Spiff on trigger

### ISSUE THREAD

**Issue by harmeet-status** [sartography/spiff-arena#1029]
Test that on "status" change of a GitHub ticket, Github actions can be configured to call SpiffWorkflow and Spiff can receive this message and **start a process model**.

**Comment by burnettk** [comment]
we can demo github-triggered events on thursday.

**Comment by madhurrya** [comment]
@burnettk could you please share the sample model you showed in the demo last week related to this. And also could you please create a sample PR for me to test this or let me know how/where to create such PR.
Also as mentioned in the other ticket does any configuration needs to be updated first to test this?

**Comment by burnettk** [comment]
we would need to update env vars on some environment in order to test that. it involves a pull request on jakub's infra repo. maybe we wait on that until @calexh-sar says he's ready for it for the bounty process?

**Comment by burnettk** [comment]
the relevant process models are:

https://dev.app.spiff.status.im/process-models/misc:bounty:bounty
https://dev.app.spiff.status.im/process-models/site-administration:github-webhook (this is the one that needs an env var to point to it)

**Comment by madhurrya** [comment]
Ok Thanks KB. Then please let me know when things are ready to be tested.

**Comment by burnettk** [comment]
we have filed https://github.com/status-im/infra-spiff-workflow/pull/41 related to this.

**Comment by madhurrya** [comment]
@burnettk can we test this now? If so please let me know where I can create PRs for that and how it'll work.

**Comment by burnettk** [comment]
sure. it's now ready. you go to https://dev.app.spiff.status.im/process-models/misc:bounty:bounty, start an instance, enter a PR url like https://github.com/sartography/spiff-arena/pull/1284. that instance should be stuck waiting. then go approve that PR in github. we should get a delivery registered at https://github.com/sartography/spiff-arena/settings/hooks/469145344?tab=deliveries and the bounty instance that was stuck should become unstuck.

**Comment by madhurrya** [comment]
Thank KB.
It seems the PR you gave about has already been approved :-(
Could you please give me another couple of PRs which I can approve tomorrow (I won't be able to test this today)

**Comment by burnettk** [comment]
You can approve an open pr any number of times. Please feel free to create a new one if you need to for testing.

**Comment by madhurrya** [comment]
I don't see an option to approve. It is already merged and deleted.
Could you please create a test PR for me, so I can play with it. Don't want to mess things there :-)

**Comment by burnettk** [comment]
@madhurrya here you go: https://github.com/sartography/spiff-arena/pull/1320

**Comment by madhurrya** [comment]
Thanks KB.

It works as explained in the above comment.

**Comment by madhurrya** [comment]
@burnettk In the Issue description Harmeet has said 'Spiff can receive this message and start a process model.'. Do you have a sample model for the Start process model part?

**Comment by burnettk** [comment]
Yeah, I think the only time when we need to start a new process instance is at the beginning with the unauthed flow.

**Comment by madhurrya** [comment]
@burnettk So in order to mark this ticket as complete, do we need to update the https://dev.app.spiff.status.im/process-models/misc:bounty:bounty process to start another model like this https://dev.app.spiff.status.im/editor/process-models/misc:qa:unauthenticated-endpoint-support/files/unauthed-endpoint-support.bpmn ?

Since that is what is mentioned in the description.

**Comment by harmeet-status** [comment]
The bounty process has slightly changed, we no longer require to start a process model, but we need to message the existing Spiff bounty process instance. I believe that we can use "Message Intermediate Catch Event"

I'm pretty sure that we have used this functionality before with PP2 or PP3. @madhurrya does that make it clearer?

**Comment by harmeet-status** [comment]
There was a discussion that Github would be unable to identify which PR's are for bounties in which are not, hence it would always call spiff every single time a PR has been approved and it would become the responsibility of spiff to only action on PR that are related to bounties.

This is bad architecture and we decided not to go ahead with this. Instead, what we have done is ask the developer to paste the URL to the PR spiff and so therefore solves the problem.

Closing this ticket as a result.
